src/aspire/source/__init__.py

```diff
@@ -115,7 +168,31 @@ def _images(self, start=0, num=None):
         :return: A 3d volume of images of size L x L x n
 
         """
-        raise NotImplementedError('Subclasses should implement this and return an Image object')
+        raise NotImplementedError('Subclasses should implement this and return a 3d ndarray')
+
+    def eval_filters(self, im_orig, start=0, num=None):
```

Comment:
The whole point of `eval_filters` to me seems to be "ImageSource, apply your configured filters, if you will, not on your images, but on these images that I'm giving you (because they're part of an optimization loop and I'm trying to see how good my images are)".

As such, you're right that the Filter object should be responsible for doing this. However, we don't have a combined `Filter` object that takes the responsibility of multiple filters and their indices (now that `SourceFilter` has gone away). I'm opening a new issue #43  on this. My suspicion is that once we have a general-purpose `Transformer` class that maintains the association between transformation function <-> indices, that would star to clear up things.